### PR TITLE
begin switching in kitchen-salt for running the test suite

### PR DESCRIPTION
**Pull request by gtmanfred**
### What does this PR do?
Work on replacing salt-jenkins-build with kitchen-salt.

This will be used so that we are running the same environment as users for our unit tests, and so that they can replicate it.

### CODE REVIEW

**Comment by gtmanfred** [comment]
Go Go Jenkins!

**Comment by thatch45** [comment]
I think this is great, although the presence of a Gemfile kind of freaks me out, can that file be commented to specify that this is local to running the test suite? Should/Could that file be in `tests/` ?

**Comment by gtmanfred** [comment]
We cannot move the gemfile and still use bundler.

Bundler requires the Gemfile to be in the directory where you are going to run the command after the install.

I can add a comment that this is only used for running the test suite.

**Comment by gtmanfred** [comment]
We could just not use bundler, and do a gem install on the file, and use that in the test suite, but bundler seems to be pretty well established as how you should run your stuff from what I have seen in the ruby community.

File: tests/copyartifacts.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
```
Comment (s0undt3ch) [inline tests/copyartifacts.py]: Perhaps umder`tests/support`?

**Comment by thatch45** [comment]
I am ok with using a gemfile, I just think that we should make sure it is very clear that the gemfile is only used for the test suite

**Comment by gtmanfred** [comment]
looks like the tests pass, I am not sure what gpg is doing, because all my commits are verified.

**Review by cachedout — APPROVED** [review]

**Pull request merged**